websites/models_test.py

```diff
@@ -283,6 +283,28 @@ def test_website_url_path_from_metadata(missing_keys):
     assert website.url_path_from_metadata(metadata) == expected_url_path
 
 
+def test_website_url_path_from_metadata_unicode():
+    """None should be returned for a site without a starter"""
+    metadata = {
+        "course_nr": "1.1",
+        "title": "テスト ウェブサイト",
+        "term": "Fall",
+        "year": "2025",
+    }
+    starter = WebsiteStarterFactory.create(
+        config={
+            "site-url-format": "[meta:course_nr]-[meta:title]-[meta:term]-[meta:year]"
+        }
+    )
+    website = WebsiteFactory.create(starter=starter, not_published=True)
+    course_nr = "1-1" if metadata.get("course_nr") is not None else "[meta:course_nr]"
+    title = "テスト-ウェブサイト" if metadata.get("title") is not None else "[meta:title]"
+    term = "fall" if metadata.get("term") is not None else "[meta:term]"
+    year = metadata.get("year", "[meta:year]")
```

Comment:
Unless you meant to parameterize `metadata`, this can be simplified.

```suggestion
    course_nr = "1-1"
    title = "テスト-ウェブサイト"
    term = "fall"
    year = "2025"
```